src/lenskit/data/relationships.py

```diff
@@ -163,19 +184,86 @@ def pandas(self, *, attributes: str | list[str] | None = None, ids=False) -> pd.
         return tbl.to_pandas()
 
     def matrix(
-        self, *, combine: MAT_AGG | Mapping[str, MAT_AGG] | None = None
+        self, *, row_entity: str = "user", col_entity: str = "item"
+    ) -> MatrixRelationshipSet:  # pragma: nocover
+        if (row_entity, col_entity) in self.mat_rset:
+            return self.mat_rset[(row_entity, col_entity)]
+        else:
+            new_mat_rset = self._make_matrix(row_entity=row_entity, col_entity=col_entity)
+            self.mat_rset[(row_entity, col_entity)] = new_mat_rset
+            return new_mat_rset
+
+    def _make_matrix(
+        self, *, row_entity: str = "user", col_entity: str = "item"
     ) -> MatrixRelationshipSet:  # pragma: nocover
         """
         Convert this relationship set into a matrix, coalescing duplicate
         observations.
 
         Args:
-            combine:
-                The method for combining attribute values for repeated
-                relationships or interactions.  Can either be a single strategy
-                or a mapping from attribute names to combination strategies.
+            row_entity:
+                The specified row entity of the matrix
+            col_entity:
+                The specified column entity of the matrix
         """
-        raise NotImplementedError()
+        if row_entity not in self.schema.entities.keys():
+            raise FieldError(self.name, row_entity)
+
+        if col_entity not in self.schema.entities.keys():
+            raise FieldError(self.name, col_entity)
+
+        if col_entity == row_entity:
+            raise ValueError("row and column entity should not be the same")
+
+        e_dict: dict[str, str | None]
+        e_dict = {row_entity: None, col_entity: None}
+        matrix_schema = RelationshipSchema(
+            entities=e_dict,
+            interaction=self.schema.interaction,
+            repeats=self.schema.repeats,
+            attributes=self.schema.attributes,
+            remove_duplicates=self.schema.remove_duplicates,
+        )
+        new_table = self._table
+
+        if "timestamp" in new_table.column_names:
+            new_table = new_table.sort_by([("timestamp", "ascending")])
+
+        if "count" not in new_table.column_names:
+            new_table = new_table.add_column(
+                new_table.num_columns, "count", pa.array(np.ones(new_table.num_rows))
+            )
```

Comment:
For the case where we do not have a `count` column, it would be better to use a `count` aggregate, instead of creating a column of ones and summing them. It will probably be faster, and will definitely require less memory.